app/pulp/app/viewsets/task.py

```diff
@@ -0,0 +1,46 @@
+from django_filters.rest_framework import filterset
+
+from pulp.app.models import Task, Worker
+from pulp.app.serializers import TaskSerializer, WorkerSerializer
+from pulp.app.viewsets import NamedModelViewSet
+from pulp.app.viewsets.custom_filters import CharInFilter
+
+from rest_framework.decorators import detail_route
+from rest_framework.response import Response
+from rest_framework import status
+
+
+class TaskFilter(filterset.FilterSet):
+    tags = CharInFilter(name='tag__name', lookup_expr='in')
+
+    class Meta:
+        model = Task
+        fields = ['state', 'worker__name', 'tags']
+
+
+class TaskViewSet(NamedModelViewSet):
+    queryset = Task.objects.all()
+    serializer_class = TaskSerializer
+    endpoint_name = 'tasks'
+    filter_class = TaskFilter
+    http_method_names = ['get', 'options']
+
+    @detail_route(methods=['post'])
+    def cancel(self, request, pk=None):
+        task = self.get_object()
+        if task.state == Task.RUNNING or task.state == Task.WAITING:
+            # TODO call a controller method to cancel task
```

Comment:
You're right, it would make more sense to have on the model given that having viewset know about the task states seems off. 
As far as having the model notify celery, maybe Django signals would be a good fit there?
https://docs.djangoproject.com/en/dev/topics/signals/